code/procs/jobprocs.dm

```diff
@@ -396,6 +409,34 @@
 				if(SL.len > 0)
 					src.set_loc(pick(SL))
 
+			if (src.traitHolder && src.traitHolder.hasTrait("pilot"))		//Has the Pilot trait - they're drifting off-station in a pod. Note that environmental checks are not needed here.
+				var/turf/pilotSpawnLocation = null
+				
+				#ifdef UNDERWATER_MAP										//This part of the code executes only if the map is a water one.
+				while(!pilotSpawnVerif(pilotSpawnLocation))					//Trying to find a valid spawn location.
```

Comment:
```suggestion
				while(!istype(pilotSpawnLocation, /turf/space))					//Trying to find a valid spawn location.
```
This does essentially the same thing since /turf/space/fluid is a subtype of /turf/space and there's not gonna be any actual space on a water map. Then you can also delete the proc above.

Reply:
Was allowed to keep the old code.

Reply:
Done.